src/openms/source/FORMAT/PercolatorInfile.cpp

```diff
@@ -130,12 +160,23 @@ namespace OpenMS
         throw Exception::ParseError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Error: line " + String(i) + " of file '" + pin_file + "' does not have the same number of columns as the header!", String(i));
       }
 
+      if (file_name_column_index >= 0)
+      {
+        raw_file_name = row[file_name_column_index];
+        if (map_filename_to_idx.find(raw_file_name) == map_filename_to_idx.end())
+        {
+          filenames.push_back(raw_file_name);
+          map_filename_to_idx[raw_file_name] = filenames.size() - 1;
+        }
+      }   
+
       const String& sSpecId = row[to_idx.at("SpecId")];
       if (sSpecId != spec_id)
       {
         pids.resize(pids.size() + 1);
         pids.back().setHigherScoreBetter(higher_score_better);
         pids.back().setScoreType(score_name);
+        pids.back().setMetaValue("map_index", map_filename_to_idx.at(raw_file_name));
```

Comment:
aah thanks. It worked with IDRipper so I assumed it is fine to use map_index.

Reply:
Yes, I think that is only a fallback. This index is needed since you can have a consensusmap of merged ids. So you need two indices for the origin (unless you start storing origins on the consensus level).